website/project/tasks.py

```diff
@@ -44,7 +44,7 @@ def update_collecting_metadata(node, saved_fields):
         if node.is_public:
             update_collected_metadata(node._id)
         else:
-            if 'is_public' in saved_fields:
+            if set(saved_fields).intersection({'spam_status', 'is_deleted', 'deleted'}):
```

Comment:
What about `is_public`?

Reply:
Also, what if the node `is_public` but also `deleted`?

Reply:
Yeah sorry this wasn't really ready yet, but now I've added testing and addressed the `is_public` thing.

Reply:
Sorry, did I jump ahead on that?